tests/queries/0_stateless/02497_if_transform_strings_to_enum.sql

```diff
@@ -0,0 +1,24 @@
+SET allow_experimental_analyzer = 1;
+SET optimize_if_transform_strings_to_enum = 1;
+
+SELECT transform(number, [2, 4, 6], ['google', 'censor.net', 'yahoo'], 'other') FROM system.numbers LIMIT 10;
```

Comment:
Need to add some test with `if` and `transform` function used as arguments of other nodes.